W16 redrafting.

jdswinbank opened the pull request:
Still a work in progress: do not merge yet. PR to provide a convenient location for commenting.

https://ldm-151.lsst.io/v/draft/

mjuric commented:
On my reading list (and also pinging @RobertLuptonTheGood).

jdswinbank committed:
Fix minor typos.

frossie committed:
More comments

frossie committed:
Final draft except for further WG comments

frossie committed:
Argh LaTeX

frossie committed:
Merge pull request #21 from lsst/u/square/4

More comments

SimonKrughoff committed:
Response to comments

SimonKrughoff committed:
Merge branch 'tickets/DM-6610' into draft

Conflicts:
	sections/algorithmic_components.tex

jdswinbank committed:
Use \emph, not \em.

This de-italicises the second half of the document and, as a bonus, it gets
the italic correction right.

TallJimbo committed:
Clarifications and fixes from first pass with JDS.

mfisherlevine committed:
Almost finished first draft

TallJimbo committed:
Make hyperlinks visible.

TallJimbo committed:
Corrections from first round of RHL comments.

This only covers sections written by JFB; no attempt has been made to
address RHL's comments on other sections.

TallJimbo committed:
Fix some multiply-defined labels.

Some of the text around these references still doesn't make much
sense, but I think these are at least the right labels to reference.

TallJimbo committed:
Add cross-references for coadd scenario bullets.

This required defining a macro that seems to limit what can go in
a description environment item (it didn't like one I had with escaped
curly braces), but I think that's enough of an edge case that it's a
worthwhile tradeoff.  Someone with better latex-fu than me may be able
to fix that as well.

TallJimbo committed:
Merge branch 'tickets/DM-6941' into draft

SimonKrughoff committed:
Adds suggestions for soft prims

SimonKrughoff committed:
Minor updates after talk with Jim

SimonKrughoff committed:
Add ref cat production section

SimonKrughoff committed:
Add map and mask to glossary

SimonKrughoff committed:
Merge branch 'tickets/DM-6966' into draft

jbecla committed:
cosmetic fixes

connolly committed:
reorganized section numbering

connolly committed:
SFM intro edits

connolly committed:
ISR edits

srp3rd committed:
Fixed the following typos:

page 25: 3.1.0.11 - “if” should be “is” in: “Source measurement if for all…”
page 28: 3.2.0.25 - Remove “w” in “…of a TemplateCoadd w that is matched to…”
page 29: 3.2.0.27 - “tempalte” should be “template” in “subtraction of the tempalte image,”
page 41: 4.1.2.3 - Remove “a” from “…this baseline calls for a a routine..”
page 41: 4.1.2.3 - Correct “thie” to “the” in “(outside the scope of thie WBS)”
page 49: 5 - “Howver” should be “However” in “Howver, some part of the graph…” at the end of the paragraph.
page 58: 5.1.2 - Missing comma(s) in “In addition to updating the CalExp WCS and PhotoCalib…”
page 61: 5.2 - “subract” should be “subtract” in “but only subract the updated…”
page 80: 5.5.1 - “later” supposed to be “layer” in “From the Science Pipelines side, Multifit is implemented as two routines, mediated by the orchestration later”
page 87: 6.1 - Third bullet sentence should use “use cases” instead of “usecases”

frossie committed:
RHL comments round 1

frossie committed:
Add overview section heading and diagram

frossie committed:
More RHL comments

frossie committed:
More RHL comments

jdswinbank committed:
Merge branch 'tickets/DM-7153' into draft

jdswinbank committed:
Merge branch 'u/jbecla/201607-fixes' into draft

connolly committed:
SFM edits

TallJimbo committed:
Fix incorrect section levels.

TallJimbo committed:
Merge image resampling into convolution kernel section.

TallJimbo committed:
Rework section headings for artifact detection.

TallJimbo committed:
Split single-visit and coadd background estimation.

These may be done on very different scales, and hence we shouldn't
assume we'll use the same algorithm (though we may).

TallJimbo committed:
Add missing section heading.

TallJimbo committed:
WF sensors are processed by T&S, not Systems Engineering.

TallJimbo committed:
Add note on DCR and JointCal.

TallJimbo committed:
Merge branch 'tickets/DM-7392' into draft

TallJimbo committed:
Add text for reference catalogs in algorithmic components.

TallJimbo committed:
Replace double quotes with form preferred by latex.

TallJimbo committed:
Add text for Warped Image Artifact Detection.

TallJimbo committed:
Add text for artifact interpolation.

TallJimbo committed:
Add text for detection algorithmic component.

connolly committed:
Alert Distribution edits

connolly committed:
MOPS edits

connolly committed:
Naming conventions

connolly committed:
Zeljko comments

connolly committed:
more of Zeljko comments

connolly committed:
more of Zeljko comments

connolly committed:
layout

frossie committed:
Right figures this time

frossie committed:
Brief introduction

frossie committed:
Fixed terminology

frossie committed:
Minor conflict resolved

timj committed:
Fix the build and use \emph not \em

Use \ref not \figure

TallJimbo committed:
Add introduction to algorithmic components sections.

frossie committed:
Switched terminology to QC

frossie committed:
Changed non-menclature to QC where appropriate

frossie committed:
Changed non-menclature to QC where appropriate and clarified focus

frossie committed:
Merge branch 'u/square/5' into draft

timj committed:
Fix latex errors

frossie committed:
More QA->QC

frossie committed:
More QA->QC

connolly committed:
Figures updated

TallJimbo committed:
Introduction for deblending algorithmic component.

TallJimbo committed:
Rework sections on joint photometric fitting.

Split joint photometric fitting into three components, tract-level,
global, and an interim global fit that includes some functionality
we think the CPP will probably cover.

TallJimbo committed:
First draft of text for deblender algorithmic component.

TallJimbo committed:
Intro for measurement section.

TallJimbo committed:
Add text for centroid algorithms.

TallJimbo committed:
Merge branch 'tickets/DM-7424' into draft

TallJimbo committed:
Add text on pixel flags and shape measurements.

mfisherlevine committed:
Starting applying RHLs comments

mfisherlevine committed:
More applying RHLs comments

mfisherlevine committed:
Finished ISR section in algorithmic componenets. Only CPP remains to be finished.

mfisherlevine committed:
Merged DM-6348, another to follow by the end of the day

connolly committed:
Edits of SFM and Distribution

mfisherlevine committed:
Added figure

mfisherlevine committed:
Added reference

mfisherlevine committed:
Cant remember!

mfisherlevine committed:
All substance done, going to debulletify a bit

TallJimbo committed:
Merge branch 'tickets/DM-6945' into draft

mfisherlevine committed:
Finished applying RHL comments, and tidying up text

mfisherlevine committed:
Merge branch 'tickets/DM-6348' into draft

connolly committed:
Edits of SSO

connolly committed:
merge of draft-ajc

connolly committed:
edits to alert generation

connolly committed:
fixed \DIA issues

jdswinbank committed:
Fix figure filename.

And thereby fix the build.

jdswinbank committed:
Fix section name.

jdswinbank committed:
Split Global Fitting section.

We have two deliverables which we need to track separately here, so I need
them to have separate sections rather than one being a sub of the other.

TallJimbo committed:
Add text for aperture fluxes.

TallJimbo committed:
Text for PSF, Kron fluxes; to-do notes on others.

jdswinbank committed:
Typo.

jdswinbank committed:
Add missing link.

TallJimbo committed:
Text for galaxy and moving point-source models.

TallJimbo committed:
Add text to blended measurement sections.

TallJimbo committed:
Merge pull request #28 from lsst/tickets/DM-6945

More updates to algorithmic components

ivezic committed:
added coauthors

ivezic committed:
minor edits (typos, etc)

ivezic committed:
Merge pull request #29 from lsst/draft-zi

Draft zi

connolly committed:
edits and refs

connolly committed:
Merge branch 'draft' of https://github.com/lsst/LDM-151 into draft

ivezic committed:
minor edits

ivezic committed:
minor edits

ivezic committed:
added new text from IPAC

ivezic committed:
Merge branch 'draft' into draft-zi

ivezic committed:
Merge pull request #30 from lsst/draft-zi

Draft zi

TallJimbo committed:
Explain 'decorrelated images' the first time it is used.

TallJimbo committed:
Clarify expected difficulty of background matching.

TallJimbo committed:
Clarify "relative simplicity" of coadd processing.

TallJimbo committed:
Clarify non-inclusion of Sources in Objects.

TallJimbo committed:
Add figure showing patch overlap regions.

TallJimbo committed:
Add figure illustrating blend chains.

TallJimbo committed:
Add reference from DRP to AP MOPS.

TallJimbo committed:
Clarify full-DRP sequence point in MOPS.

TallJimbo committed:
Decrease alarm about sequence points in classification.

TallJimbo committed:
Include CPP outputs in list of DRP inputs.

ivezic committed:
Merge pull request #31 from lsst/tickets/DM-8496

DM-8496: Updates to DRP section from review

frossie committed:
Some QA->QC corrections that slipped through.

Commit:
Rename DIAExp to DiffExp

DIAExp is used in the DRP section for the Difference Images,
while DiffExp is used in the AP section.  Rename them all to
DiffExp to be consistent.

Commit:
Modify the production DAG definition

Pipeline as edges cannot work.

Commit:
Correct typo: March -> Match

Commit:
Correct L2 data product table in persisting best seeing coadds

LSE-61 (draft 2016/12/02) Persisting Data Products section
requires both deep coadds and best seeing coadds to be
archived rather than regenerated on demand.

mfisherlevine committed:
Added a new command

mfisherlevine committed:
Lots of changes, work is for tickets DM-9131, DM-8066 and DM-9132. not yet finsihed though.

Commit:
Rename EpochRangeCoadd to ShortPeriodCoadd in L2 data product Table

Rename EpochRangeCoadd in Table 2 to be consistent with the texts
and DPDD.

mfisherlevine committed:
Added some references

mfisherlevine committed:
Finished applying comments and adding section. Sending to ZI and RHL for comments

Commit:
Add TemplateCoadd to L2 Data Product Table

TemplateCoadd is listed as a L1 Image Product in DPDD (4.4.3)

hsinfang committed:
Merge pull request #32 from lsst/tickets/DM-9345

DM-9345 Minor corrections and renaming for consistency

ebellm committed:
Reduce level of detail in several sections and respond to RHL comments.

Reduce algorithmic detail specified in:

   PSF Estimation - Single CCD PSF Estimation
   Astrometric Fitting - Single CCD
   PSF Matching -Image Subtraction
   Association and Matching - Single CCD to Reference Catalog, Semi-Blind
   Association and Matching - Single Visit to Reference Catalog, Semi-Blind

Respond to RHL comments throughout these sections.

Also various typo and reference fixes.

mfisherlevine committed:
Merge branch 'draft' into tickets/DM-8066

mfisherlevine committed:
Merge pull request #34 from lsst/tickets/DM-8066

Tickets/dm 8066

timj committed:
Move blended measurement paper to DMTN-038

timj committed:
Merge pull request #35 from lsst/tickets/DM-9288

Move blended measurement paper to DMTN-038

Commit:
Initial pass extracting references from LDM-151

timj committed:
Switch document to bibtex

Still some missing references.

timj committed:
Travis needs the bst file

timj committed:
Switch to lsst doc latex class file

timj committed:
Add missing refernces following review

timj committed:
Fix Kubica reference

timj committed:
Merge pull request #36 from lsst/tickets/DM-6574

DM-6574: Switch to bibtex

timj committed:
Use more modern distribution for travis and use lsst-texmf

Required for new LSST style

timj committed:
Add instructions for building the PDF to the README

timj committed:
Rename to LDM-151.tex

No longer need the intermediate step as the new latex style
includes the cover page directly.

timj committed:
Fix duplicate section labels

timj committed:
Remove trailing whitespace

timj committed:
Fix Kubica reference

timj committed:
Add non-breakable space for author names

timj committed:
Merge pull request #37 from lsst/tickets/DM-9975

DM-9975: Switch to LSST latex class

timj committed:
Cite LSST documents explicitly

They now appear in the references.

timj committed:
Add missing \DPDD and \SRD

jdswinbank committed:
Clarify description of flat-fielding.

jdswinbank committed:
Drop unused aasjournal.bst.

timj committed:
Merge pull request #38 from lsst/tickets/DM-10047

Cite LSST documents explicitly

jdswinbank committed:
Merge branch 'tickets/DM-10197' into draft

ebellm committed:
Minor revisions in response to github reviews.

ebellm committed:
resolve merge conflicts

ebellm committed:
Merge branch 'tickets/DM-9624' into draft

Minor cleanup from github reviewer comments.

timj committed:
Fix LDM-227 reference usage

timj committed:
Use modern citeds rather than citell

timj committed:
Merge pull request #39 from lsst/tickets/DM-10047-fixups

DM-10047: Correct the LDM-227 reference

timj committed:
Remove spurious tilde from end of section declarations

timj committed:
Remove un-needed \\ spacing between paragraphs

timj committed:
Fix double quotes

timj committed:
Modernize \em, \bf, \it, \tt, \rm

timj committed:
Use input rather than include

\include adds a page break but we do not want a page break after
the preamble as the preamble has already included one.

timj committed:
Merge pull request #40 from lsst/u/timj/spurious-spacing

Clean up some weird spacing, quotes and latex2.09

timj committed:
Properly cite DMTN documents

timj committed:
Reference LDM documents mentioned in figure 1

timj committed:
Merge pull request #41 from lsst/u/timj/dmtn

Properly cite DMTN documents

timj committed:
Remove SUIT section

timj committed:
Remove QA section

TallJimbo committed:
Add a bit of clarifying text to some Algorithmic Components

TallJimbo committed:
Remove outline from section with text.

TallJimbo committed:
More comments on algorithms.

TallJimbo committed:
Add text for full-visit PSF estimation.

TallJimbo committed:
Add text for aperture corrections.

TallJimbo committed:
Add text on image coaddition algorithm.

timj committed:
Minor bib file cleanups

Remove some unused entries and remove an entry that is now
in the shared database.

timj committed:
Merge pull request #43 from lsst/u/timj/fix-bibs

Minor bib file cleanups

timj committed:
Fix Galamhos citation

Broken by mistake in early commit that cleaned up the Galamhos
entry.

timj committed:
Fix some spelling mistakes

timj committed:
Merge pull request #44 from lsst/u/timj/fix-galamhos

Fix Galamhos usage and some spelling

timj committed:
Stop referring to SDQA section in intro

ebellm committed:
Revise Level 1 flowchart figures.

Replace Level 1 overview figure and subsidiaries (SFM, Alert Generation,
Alert Distribution).

Remove unused (and outdated) jpeg version.

TallJimbo committed:
Add text for S/G classification.

SimonKrughoff committed:
Updates to algorithmics components

This also inludes a small update to remove a reference to a missing
table.  I just replaced it with a reference to the DPDD.

Most refactoring here was removing specific algorithmic details.  Where
references to relevant algorithms exist in the community, I left them
in.

TallJimbo committed:
Add text on matching and measurement calibration.

TallJimbo committed:
Add text for coadd decorrelation.

TallJimbo committed:
Add comment on what to do with Software Primitives.

ebellm committed:
Merge branch 'tickets/DM-10003' into draft

SimonKrughoff committed:
Format common objects

mfisherlevine committed:
Update aux telescope processing

jdswinbank committed:
Merge pull request #47 from lsst/tickets/DM-9361

Update aux telescope processing

SimonKrughoff committed:
Fix some references

SimonKrughoff committed:
Merge branch 'tickets/DM-9626' into draft

timj committed:
Add ref to DMTN-037

timj committed:
Remove trailing whitespace

timj committed:
Merge pull request #48 from lsst/u/timj/fix-dmtn37-cite

Cite DMTN-037 rather than using a URL

timj committed:
texmf now needs texlive-math-extra

mjuric committed:
Merge branch 'tickets/DM-8497' into draft

mjuric committed:
Merge remote-tracking branch 'origin/tickets/DM-9296' into draft

mjuric committed:
Rename the document to "DM Science Pipelines Design"

mjuric committed:
Remove outdated reference to the UML model

timj committed:
Remove irrelevant commentary

timj committed:
Add pgf dependency

Needed by tcolorbox (but lsst-texmf installs it via latex-beamer)

mjuric committed:
introduction.tex: use the LSST Science Platform terminology

mjuric committed:
preface.tex: remove reference to the UML model, emphasize Agile

mjuric committed:
introduction.tex: data access and software availability clarifications

mjuric committed:
introduction.tex: avoid using the term 'Applications'

mjuric committed:
alert_production_pipelines.tex: table placement fixups

mjuric committed:
alert_production_pipelines.tex: figure placement fix

mjuric committed:
alert_production_pipelines.tex: clean up alert numbers and requirement wording

mjuric committed:
.gitignore: add *~ pattern

mjuric committed:
calibration_products.tex: vanquish \dragons

mjuric committed:
Makefile: depend on sections/calibration_products.tex

mjuric committed:
data_release_production_pipelines.tex: fix table placement

mjuric committed:
algorithmic_components.tex: add reference to SubtractSnaps from DRP

mjuric committed:
software_primitives.tex: state we may move this to MagicDraw

mjuric committed:
LDM-151.tex: Introduce \ifannotated mechanism to hide annotations

mjuric committed:
software_primitives.tex: Hide the Tools section behind \ifannotated

mjuric committed:
alert_production_pipelines.tex: remove XXX (CAVEAT: I think this paragrah may not be right)

mjuric committed:
algorithmic_components.tex: fix incorrect label references

timj committed:
Install full texlive

timj committed:
Merge pull request #49 from lsst/u/timj/check-build

Install full texlive

mjuric committed:
Added co-authors: Bellm, Chiang, Pietrowicz, Reiss and Yoachim

mjuric committed:
Make the Pipelines Scientist (Lupton) the doc. owner and lead author

mjuric committed:
LDM-151.tex: Reduce tocdepth to 3

timj commented on sections/alert_production_pipelines.tex:
```diff
@@ -0,0 +1,510 @@
+\subsubsection{Input Data}
+\label{sec:apSFMinput}
+
+\paragraph*{Raw Camera Images:} Amplifier images that have been corrected for crosstalk and bias by the camera software. All images from a visit should be available to the task (including snaps). An approximate WCS is assumed to be available as metadata derived from the Telescope Control System with an absolute pointing uncertainty (for a full focal plane) of 2 arcseconds (OSS-REQ-0298) and the f ...
```
Where does the 32 arcseconds come from in LTS-206? LTS-206 is a bit problematic because it doesn't have requirements numbers and parameters like this aren't explicitly called out. I see section 3.4.3 say that the rotator encoder accuracy is 0.002 degrees (stated as one pixel at the edge of the focal plane) and 3.4.2 states the absolute accuracy is 0.009 degrees (which I believe is the 32 arcseconds). It's somewhat confusing that the number in LDM-151 can not be found directly in LTS-206.

SimonKrughoff committed:
Fix typo in algo comp

timj committed:
Refer to explicit OSS parameters and requirements

Where already mentioned in the text.

timj committed:
Merge pull request #51 from lsst/u/timj/oss-refs

Refer to explicit OSS parameters and requirements

timj commented on sections/calibration_products.tex:
```diff
@@ -0,0 +1,659 @@
+The work performed in this WBS serves several complementary roles:
+
+\begin{itemize}
+ \item It will enable the production of calibration data products as required by the Level 2 Photometric Calibration Plan (\NewPCP{}) and other planning documents \cite{Lupton15}. This includes both characterization of the sensitivity of the LSST system (optics, filters and detector) and the transmissivity of, and emission from, the atmosphere;
```
We should not be citing Lupton15. We should decide whether this document should go on DocuShare or whether this document is no longer needed.

mfisherlevine replied:
OK. I suggest we just remove the citation for now then.

timj commented on sections/calibration_products.tex:
```diff
@@ -0,0 +1,659 @@
+
+
+\subsubsection{Filter Transmission}\label{sec:CPP:output:filterTransmission}
+Monitoring of filter transmission \emph{in-situ}. As well as the filter transmission measurement provided by the camera team/vendor in \secsymbol\ref{sec:CPP:inputs:filterTransmission}, we further baseline the development of a procedure for monitoring the filter response at 1\,nm resolution using the approach described in \cite{Lupton15}, \ie by making suitable CBP measurements with and without t ...
```
Fix Lupton15 mention.

mfisherlevine replied:
OK, will remove here too then,

timj commented on sections/calibration_products.tex:
```diff
@@ -0,0 +1,659 @@
+
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
@mfisherlevine Why is this commented out text in this document?

mfisherlevine replied:
Oh, I think that was just the text that was in there waaaaay back in the day before I started work on this at all, so I just left it there while the work was ongoing. It's now vestigial and I will remove it.

mfisherlevine committed:
Remove citations of Lupton15

mfisherlevine committed:
Remove vestigial comments

Before the overhaul of this document some text existed for this section.
It wasn't deleted, but was just commented out and pushed to the bottom.
This commit removes that old text.

mfisherlevine committed:
Remove other irrelevant comments

mfisherlevine committed:
Whitespace cleanup

mfisherlevine committed:
Merge pull request #53 from lsst/tickets/DM-10511

DM-10511: Remove references to unreleased calibration document

timj commented:
When this PR is approved I will do the merge to ensure that the change record is correct.

timj commented:
What's the plan for these:
```
LaTeX Warning: Hyper reference `sec:acWarping' on page 75 undefined on input li
LaTeX Warning: Hyper reference `sec:acWarping' on page 78 undefined on input li
LaTeX Warning: Hyper reference `sec:acWarping' on page 78 undefined on input li
LaTeX Warning: Hyper reference `sec:acWarping' on page 80 undefined on input li
LaTeX Warning: Hyper reference `sec:acWarping' on page 81 undefined on input li
LaTeX Warning: Hyper reference `sec:acWarping' on page 81 undefined on input li
LaTeX Warning: Hyper reference `sec:acWarping' on page 81 undefined on input li
LaTeX Warning: Hyper reference `sec:drpMOPS' on page 84 undefined on input line
LaTeX Warning: Hyper reference `sec:drpFootprintsHeavy' on page 85 undefined on
LaTeX Warning: Hyper reference `drpDeepAssociate' on page 106 undefined on inpu
LaTeX Warning: Hyper reference `drpDeepDeblend' on page 106 undefined on input 
LaTeX Warning: Hyper reference `drpMultiFit' on page 113 undefined on input lin
LaTeX Warning: Hyper reference `sec:ForcedMeasurement' on page 113 undefined on
LaTeX Warning: Hyper reference `ac:CentroidAlgorithms' on page 115 undefined on
LaTeX Warning: Hyper reference `sec:acPsfMatching' on page 124 undefined on inp
LaTeX Warning: Hyper reference `sec:spWarping' on page 139 undefined on input l
LaTeX Warning: Hyper reference `spTablesNWayMatching' on page 149 undefined on 
```
Given that they result in no-ops in the output PDF (no link is put in) they can all be removed. I'd rather that then add "undefined reference" warnings to the output.

RobertLuptonTheGood commented:
I assume that they are all/mostly typos for real cross references.  I think most are @TallJimbo's and he needs to go through and fix them.

mjuric committed:
LDM-151.tex: Make J. Swinbank the lead author

Per DMLT F2F discussion on 2017-05-11, John Swinbank -- as the manager to be
responsible for science pipeline implementation -- will be the owner of this
document.

mjuric committed:
LDM-151.tex: Re-sort authors alphabetically (except for owner)

mjuric committed:
LDM-151.tex: Remove S.R.~Pietrowicz from author list

Did not respond to requests to confirm he wishes to be included as a
co-author.

timj committed:
Remove spurious mention of LDM-151 in footnote

With new document class there is no longer a doubt that this document
is LDM-151.

timj committed:
References have moved to shared bib file

timj committed:
Remove trailing whitespace

jdswinbank committed:
Set document curator.

jdswinbank committed:
Remove data products figure.

This doesn't add any information beyond the text, and may be actively
confusing.

jdswinbank committed:
Fix broken references.

jdswinbank committed:
Merge branch 'tickets/DM-10538' into draft

timj committed:
Update change record and date to prepare for approved release
